VideoLocker/src/main/java/org/edx/mobile/task/Task.java

```diff
@@ -21,19 +23,26 @@
 
 public abstract class Task<T> extends RoboAsyncTask<T> {
 
-    private ProgressBar progressBar;
+    public enum Type {
+        USER_INITIATED, LOADING_CACHED, LOADING_NON_CACHED
+    }
+
+    protected final Handler handler = new Handler();
+    protected final Logger logger = new Logger(getClass().getName());
 
     @Nullable
     private WeakReference<TaskProgressCallback> progressCallback;
+
     @Nullable
     private WeakReference<TaskMessageCallback> messageCallback;
 
-    protected final Handler handler = new Handler();
-    protected final Logger logger = new Logger(getClass().getName());
+    private ProgressBar progressBar;
 
     @Inject
     protected IEdxEnvironment environment;
 
+    private Type taskType = Type.LOADING_NON_CACHED;
+
     public Task(Context context) {
         super(context);
```

Comment:
How about calling `this(context, Type.LOADING_NON_CACHED);` from here, making `taskType` `final`, and centralizing the initialization logic in one constructor?